aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -218,6 +218,9 @@ Tensor& normal_out_impl(Tensor& output, const Tensor& mean, double std, c10::opt
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, double mean, const Tensor& std, c10::optional<Generator> gen) {
   TORCH_CHECK(!std.is_complex(), "normal expects standard deviation to be non-complex");
+  TORCH_CHECK(
+    std.min().gt(0).item<bool>(),
```

Comment:
not introduced by this PR, but why don't we allow 0 std? numpy allows it.

Reply:
That's a good point. It seems like `torch.normal` already has the logic to handle std=0. Before this PR, this gives the correct result:

```python
>>> torch.normal(torch.tensor([1., 2., 0.]), torch.tensor([0.]))
tensor([1., 2., 0.])
```

I think it would be a good idea to allow std=0. Should I go ahead and do that?

Reply:
Yes please, and also can you enable it for scalar std (it fails currently):
```
In [2]: a=torch.empty(3, device="cuda").normal_(3,0)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-3a09ad3edf5c> in <module>
----> 1 a=torch.empty(3, device="cuda").normal_(3,0)

RuntimeError: normal_ expects std > 0.0, but found std=0

In [3]: torch.normal(torch.tensor([1., 2., 0.]), 0)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-3-ce789585013c> in <module>
----> 1 torch.normal(torch.tensor([1., 2., 0.]), 0)

RuntimeError: normal_ expects std > 0.0, but found std=0

```